flepimop/gempyor_pkg/src/gempyor/outcomes.py

```diff
@@ -124,6 +118,34 @@ def onerun_delayframe_outcomes(
         )
 
 
+def _onerun_delayframe_outcomes_with_random_seed(
+    random_seed: int,
+    sim_id2write: int,
+    modinf: model_info.ModelInfo,
+    load_ID: bool = False,
+    sim_id2load: int = None,
+) -> None:
+    """
+    Wrapper function to run `onerun_delayframe_outcomes` with a random seed.
+
+    Args:
+        random_seed: Random seed to use for the run.
+        sim_id2write: Simulation ID to write.
+        modinf: ModelInfo object.
+
+    Returns:
+        None
+
+    See Also:
+        `onerun_delayframe_outcomes`
+
+    """
+    np.random.seed(seed=random_seed)
+    onerun_delayframe_outcomes(
+        sim_id2write, modinf, load_ID=load_ID, sim_id2load=sim_id2load
+    )
```

Comment:
This solution seems fine for now.

However, I foresee that we may have this need more generically: that is, to have a core function which can run with or without a random number set. This seems like an ideal use case for decorating with some sort of intercepting wrapper that will set the random seed if provided, otherwise proceed as normal

Reply:
I do not foresee this. Global seed setting with `numpy.random.seed` is considered legacy, see: https://numpy.org/doc/2.2/reference/random/generated/numpy.random.seed.html. Instead we should move to providing an explicit `numpy.random.Generator` in the future, see: https://github.com/HopkinsIDD/flepiMoP/pull/429#issuecomment-2590013488.

Reply:
Hmm. If not passing seed, then passing generator vs using "global" one. Should probably be that we're writing things in the other direction then - the explicit generator-passed-version is the real method, and the wrapping method is a version that will fill in the global generator (and probably issue some sort of warning)?

Relative to implementating distributions - my instinct is that we want *our* distribution classes to have our draw methods receive a generator instance. This is a bit more overarching random number generation overhaul, however.